core/embed/rust/src/ui/model_mercury/flow/confirm_output.rs

```diff
@@ -212,61 +210,30 @@ fn get_cancel_page(
     })
 }
 
-#[allow(clippy::not_unsafe_ptr_arg_deref)]
-pub extern "C" fn new_confirm_output(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
-    unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, new_confirm_output_obj) }
-}
-
-fn new_confirm_output_obj(_args: &[Obj], kwargs: &Map) -> Result<Obj, error::Error> {
-    let title: Option<TString> = kwargs.get(Qstr::MP_QSTR_title)?.try_into_option()?;
-    let subtitle: Option<TString> = kwargs.get(Qstr::MP_QSTR_subtitle)?.try_into_option()?;
-
-    let account: Option<TString> = kwargs.get(Qstr::MP_QSTR_account)?.try_into_option()?;
-    let account_path: Option<TString> =
-        kwargs.get(Qstr::MP_QSTR_account_path)?.try_into_option()?;
-
-    let br_name: TString = kwargs.get(Qstr::MP_QSTR_br_name)?.try_into()?;
-    let br_code: u16 = kwargs.get(Qstr::MP_QSTR_br_code)?.try_into()?;
-
-    let message: Obj = kwargs.get(Qstr::MP_QSTR_message)?;
-    let amount: Option<Obj> = kwargs.get(Qstr::MP_QSTR_amount)?.try_into_option()?;
-
-    let chunkify: bool = kwargs.get_or(Qstr::MP_QSTR_chunkify, false)?;
-    let text_mono: bool = kwargs.get_or(Qstr::MP_QSTR_text_mono, true)?;
-
-    let address: Option<Obj> = kwargs.get(Qstr::MP_QSTR_address)?.try_into_option()?;
-    let address_title: Option<TString> =
-        kwargs.get(Qstr::MP_QSTR_address_title)?.try_into_option()?;
-
-    let summary_items: Obj = kwargs.get(Qstr::MP_QSTR_summary_items)?;
-    let fee_items: Obj = kwargs.get(Qstr::MP_QSTR_fee_items)?;
-
-    let summary_title: Option<TString> =
-        kwargs.get(Qstr::MP_QSTR_summary_title)?.try_into_option()?;
-    let summary_br_name: Option<TString> = kwargs
-        .get(Qstr::MP_QSTR_summary_br_name)?
-        .try_into_option()?;
-    let summary_br_code: Option<u16> = kwargs
-        .get(Qstr::MP_QSTR_summary_br_code)?
-        .try_into_option()?;
-
-    let cancel_text: Option<TString> = kwargs.get(Qstr::MP_QSTR_cancel_text)?.try_into_option()?;
-
+pub fn new_confirm_output(
```

Comment:
Other possibility is to use `ConfirmOutput::new` at the cost of additional indent. Or module-level `fn new` if there's just one flow.

Also how is clippy not complaining about argument count? I suppose it's because clippy ignores all UI code that is not model T?

Reply:
Rationale for using standalone functions instead of putting it to `impl X` is to be consistent across all flows. Some flows have multiple enums defined (various number of states) and then it's not clear to which `impl X/Y/Z` to put it.